packages/labs/analyzer/src/lib/lit-element/lit-element.ts

```diff
@@ -33,18 +39,75 @@ export const getLitElementDeclaration = (
   declaration: LitClassDeclaration,
   analyzer: AnalyzerInterface
 ): LitElementDeclaration => {
+  const reactiveProperties = getProperties(declaration, analyzer);
   return new LitElementDeclaration({
     tagname: getTagName(analyzer.typescript, declaration),
     // TODO(kschaaf): support anonymous class expressions when assigned to a const
     name: declaration.name?.text ?? '',
     node: declaration,
-    reactiveProperties: getProperties(declaration, analyzer),
+    reactiveProperties,
     ...getJSDocData(declaration, analyzer),
     getHeritage: () => getHeritage(declaration, analyzer),
-    ...getClassMembers(declaration, analyzer),
+    ...getLitElementClassMembers(declaration, analyzer, reactiveProperties),
   });
 };
 
+function getCustomElementField(
+  field: ClassField,
+  prop: ReactiveProperty
+): CustomElementField {
+  const {name} = field;
+  const {
+    privacy,
+    inheritedFrom,
+    source,
+    readonly,
+    type,
+    description,
+    deprecated,
+    summary,
+  } = field;
+  const attribute =
```

Comment:
I wonder how well the current code handles property metadata merging? ie, you can have `@property()` for the same field on a base class and sub class and the sub class will get merged property options.

Reply:
I've lost track over the holidays, last eng meeting i came to we agreed not to do comprehensive inheritance inside the analyzer class, but rather provide tools (which presumably would be responsible for disk io and package resolution)

How does that affect this thread?